### PR TITLE
Make Tide ignore pending batch jobs containing PRs that have left the pool of been updated.

### LINKED ISSUE #9558: tide dashboard bug:
screen shot 2018-09-24 at 8 38 11 pm

### PR DESCRIPTION
**Pull request by cjwagner**
fixes #9558 
I tried to make this a little more readable as well.

/cc @BenTheElder @stevekuznetsov 
/kind bug

### CODE REVIEW

File: prow/tide/tide.go [hunk 1]
```diff
@@ -551,45 +551,46 @@ func accumulateBatch(presubmits map[int]sets.String, prs []PullRequest, pjs []ku
...
-			}
-			log.Debugf("no new batch necessary, current batch pending: %v", pendingNums)
-			return nil, pending
-		}
-		// Otherwise, accumulate results.
+		// First validate the batch job's refs.
 		ref := pj.Spec.Refs.String()
-		if _, ok := states[ref]; !ok {
-			states[ref] = &accState{
+		state, ok := states[ref]
```
Comment (stevekuznetsov) [inline prow/tide/tide.go]: This is a little hard to follow. I think the `if !ok` block is enforcing that we do have `states[ref]` ultimately, we should

```go
if _, ok := states[ref]; !ok {
    state := &accState{
    // ...
    states[ref] = state
}
```

And always then use `states[ref]` in the following lines -- easier to reason about what is in the `state` variable

Comment (cjwagner) [reply prow/tide/tide.go]: I made this change for consistency with the loop at the end of the function and because I thought it was confusing that we were repeatedly looking up the same value from the map every time we wanted to edit it instead of reusing the value. It is unclear to me why repeating the map lookup everywhere would make things any easier to reason about.

I switched back to repeating the map lookup everywhere though so that we can get this fix PR merged.

File: prow/tide/tide.go [hunk 2]
```diff
@@ -551,45 +551,46 @@ func accumulateBatch(presubmits map[int]sets.String, prs []PullRequest, pjs []ku
...
+
+		// Accumulate job states by batch ref.
 		context := pj.Spec.Context
-		if s, ok := states[ref].jobStates[context]; !ok || s == noneState {
-			states[ref].jobStates[context] = toSimpleState(pj.Status.State)
+		if s, ok := state.jobStates[context]; !ok || s == noneState {
+			state.jobStates[context] = jobState
 		}
 	}
 	for ref, state := range states {
```
Comment (cjwagner) [inline prow/tide/tide.go]: @stevekuznetsov Based on your comment above this should to be switched to `for ref := range states {` and use `states[ref]` instead of `state`?

**Comment by stevekuznetsov** [comment]
Oh, I see. Sorry Cole, that part of the func was not in the diff so I didn't see it at all and the changes you had were confusing. It makes more sense now.

This is LGTM and if you'd like to revert to the other approach we can merge that too, I see the value in it now.

/lgtm
